core/src/main/java/org/mule/config/pool/DefaultThreadPoolFactory.java

```diff
@@ -28,19 +30,20 @@
 
     public ThreadPoolExecutor createPool(String name, ThreadingProfile tp)
     {
+        ThreadingProfile threadingProfileCopy = tp instanceof DirectThreadingProfile ? tp : new ImmutableThreadingProfile(tp);
```

Comment:
Is not good to have "instanceof DirectThreadingProfile" in different places. I'm not asking for a shared method to check that, is just that we need to remind to add this check when we add new threading profile related code